src/main/java/roomescape/controller/ReservationApiController.java

```diff
@@ -0,0 +1,69 @@
+package roomescape.controller;
+
+import jakarta.validation.Valid;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLong;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import roomescape.domain.Reservation;
+import roomescape.exception.ErrorMessage;
+import roomescape.exception.NotFoundReservationException;
+
+@RestController // @Controller + @ResponseBody 특정 클래스가 RESTful 웹 서비스의 컨트롤러 역할을 하도록 지정한다
+public class ReservationApiController {
+
+  private List<Reservation> reservations = new ArrayList<>();
+  private AtomicLong index = new AtomicLong(1);
+
+  @GetMapping("/reservations")
+  public ResponseEntity<List<Reservation>> readReservation() {
+    return ResponseEntity.ok().body(reservations);
+    // GET /reservations 요청이 들어왔을 때 저장된 모든 예약 정보를 반환한다
+    // ResponseEntity 는 HTTP 응답을 위한 클래스이다. ok() 메소드는 200 OK 상태 코드를 반환한다
+    // body() 메서드는 응답 본문에 들어갈 데이터를 설정한다. 반환할 데이터는 reservations 필드에 저장된 예약 정보이다
+  }
+
+  @PostMapping("/reservations")
+  public ResponseEntity<Reservation> createReservation(@Valid @RequestBody
+  Reservation reservation, BindingResult bindingResult) {
+    if (bindingResult.hasErrors()) {
+      throw new IllegalArgumentException(ErrorMessage.INVALID_ARGUMENT.getMessage());
+    }
+    Reservation newReservation
+        = Reservation.createNewReservation(index.getAndIncrement(), reservation);
+    reservations.add(newReservation);
+    return ResponseEntity.created(URI.create("/reservations/" + newReservation.getId()))
+                         .header("Content-Type", "application/json")
```

Comment:
```suggestion
                         .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
```

이렇게 enum을 사용해서 실수를 줄일 수 있을 것 같아요!

또한, body에 값을 담았기 때문에 자동으로 헤더 설정이 되어서 굳이 지정해주지 않아도 됩니다!

Reply:
body에 값을 담으면 자동으로 헤더가 설정되는군요. 감사합니다!